flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -325,11 +325,15 @@ private void processMessage(Delivery delivery, RMQCollectorImpl collector) throw
     @Override
     public void run(SourceContext<OUT> ctx) throws Exception {
         final RMQCollectorImpl collector = new RMQCollectorImpl(ctx);
+        final long timeout = rmqConnectionConfig.getDeliveryTimeout();
         while (running) {
-            Delivery delivery = consumer.nextDelivery();
+            Delivery delivery =
+                    timeout > 0 ? consumer.nextDelivery(timeout) : consumer.nextDelivery();
```

Comment:
Can the `timeout` actually be below zero? In general, I would not allow calling `consumer.nextDelivery()` at all because it prevents the job from making progress.

Reply:
It can be set to zero. Zero is the current, problematic, behavior (no timeout). Any positive number will enable the timeout (and it will be enabled by default). Do you mean, the "no timeout" case should be dropped completely?

Reply:
IMO we should change the validation slightly to always require a value greater than 0. I assume that most RabbitMQ uses cases are not very performance-critical and can easily take a small penalty.

Preventing the bug from happening is the highest priority.